### PR TITLE
Hooks: Extract: always extract to $pkgname-$version

### PR DESCRIPTION
**Pull request by sgn**
#### Testing the changes
- I tested the changes in this PR: **briefly**

Close #33013 Direction for #40082 #40101 #40071 

[ci skip] [skip ci]

### CODE REVIEW

File: common/environment/setup/sourcepkg.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 ## VARIABLES
 unset -v pkgname version revision short_desc homepage license maintainer
 unset -v archs distfiles checksum build_style build_helper nocross broken
-unset -v configure_script configure_args wrksrc build_wrksrc create_wrksrc
```
Comment (classabbyamp) [inline common/environment/setup/sourcepkg.sh:7]: this should go in a different commit i think?

**Comment by sgn** [comment]
I'll keep it here until python3 rebuild completed #39346

File: common/environment/setup/sourcepkg.sh (hunk 1 above)
Comment (sgn) [reply common/environment/setup/sourcepkg.sh:7]: Maybe, but this commit is very small, so I don't know.

File: common/hooks/do-extract/00-distfiles.sh [hunk 2]
```diff
@@ -20,17 +20,16 @@ hook() {
...
-	fi
-
 	# Disable trap on ERR; the code is smart enough to report errors and abort.
 	trap - ERR
 
 	TAR_CMD="$(command -v bsdtar)"
 	[ -z "$TAR_CMD" ] && TAR_CMD="$(command -v tar)"
 	[ -z "$TAR_CMD" ] && msg_error "xbps-src: no suitable tar cmd (bsdtar, tar)\n"
 
+	extractdir=$(mktemp -d "$XBPS_BUILDDIR/.extractdir-XXXXXXX") ||
```
Comment (classabbyamp) [inline common/hooks/do-extract/00-distfiles.sh:30]: should the tmpdir have the pkgname in it?

File: common/hooks/do-extract/00-distfiles.sh [hunk 3]
```diff
@@ -163,4 +152,27 @@ hook() {
 			;;
 		esac
 	done
+
+	# find "$extractdir" -mindepth 1 -maxdepth 1 -printf '1\n' | wc -l
```
Comment (classabbyamp) [inline common/hooks/do-extract/00-distfiles.sh:156]: could this be done with `ls -A1 | wc -l`?

File: common/hooks/do-extract/00-distfiles.sh (hunk 2 above)
Comment (sgn) [reply common/hooks/do-extract/00-distfiles.sh:30]: Not really important, by why not :D

File: common/hooks/do-extract/00-distfiles.sh (hunk 3 above)
Comment (sgn) [reply common/hooks/do-extract/00-distfiles.sh:156]: Thinking again, I think I prefer a shell for of globs instead, `ls | wc` needs 2 `fork`s and `pipe`.

File: common/hooks/do-extract/00-distfiles.sh (hunk 2 above)
Comment (sgn) [reply common/hooks/do-extract/00-distfiles.sh:30]: I'm worry that `pkgname` may contain `XXX`, but no
> TEMPLATE must contain at least 3 consecutive 'X's in last component

File: common/hooks/do-extract/00-distfiles.sh (hunk 3 above)
Comment (sgn) [reply common/hooks/do-extract/00-distfiles.sh:156]: And we need to get the subdirs' name anyway :-p
